features/harmful-apis/main.js

```diff
@@ -0,0 +1,329 @@
+        }
+    },
+    {
+        id: 'mediadevices-enumarete-devices-api',
+        run: () => {
+            return ('MediaDevices' in window) && ('enumerateDevices' in MediaDevices.prototype);
+        }
+    },
+    {
+        id: 'storage-manager-estimate-api',
+        run: () => {
+            return ('StorageManager' in window) && ('estimate' in StorageManager.prototype);
+        }
+    },
+    {
+        id: 'network-information-api',
+        run: () => {
+            return ('connection' in navigator);
+        }
+    },
+    {
+        id: 'client-hints-api',
+        run: () => {
+            return fetch('/reflect-headers')
+                .then(r => r.json())
+                .then(data => {
+                    const chData = {
+                        'device-memory': data.headers['device-memory'],
+                        downlink: data.headers.downlink,
+                        dpr: data.headers.dpr,
+                        ect: data.headers.ect,
+                        rtt: data.headers.rtt,
+                        'viewport-width': data.headers['viewport-width']
+                    };
+
+                    if (Object.values(chData).every(i => i === undefined)) {
+                        return false;
+                    }
+
+                    return chData;
+                });
+        }
+    },
+    {
+        id: 'topics-api',
+        run: () => {
+            return ('browsingTopics' in document);
+        }
+    },
+    {
+        id: 'floc-api',
+        run: () => {
+            return ('interestCohort' in document);
+        }
+    },
+    {
+        id: 'raw-sockets-api',
+        run: () => {
+            return ('openUDPSocket' in navigator) || ('openTCPSocket' in navigator);
+        }
+    },
+    {
+        id: 'idle-detection-api',
+        run: () => {
+            return ('IdleDetector' in window);
+        }
+    },
+    {
+        id: 'raw-clipboard-api',
+        run: () => {
+            return ('clipboard' in navigator) && ('platform' in navigator.clipboard);
+        }
+    },
+    {
+        id: 'native-filesystem-api',
+        run: () => {
+            return ('showOpenFilePicker' in window);
+        }
+    },
+    {
+        id: 'background-sync-api',
+        run: () => {
+            return ('ServiceWorkerRegistration' in window) && ('sync' in ServiceWorkerRegistration.prototype);
+        }
+    },
+    {
+        id: 'periodic-sync-api',
+        run: () => {
+            return ('ServiceWorkerRegistration' in window) && ('periodicSync' in ServiceWorkerRegistration.prototype);
+        }
+    },
+    {
+        id: 'push-api',
+        run: () => {
+            return ('ServiceWorkerRegistration' in window) && ('pushManager' in ServiceWorkerRegistration.prototype);
+        }
+    },
+    {
+        id: 'first-party-sets-api',
+        run: () => {
+            if (!window.cookieStore) {
+                return '❌ Can\'t be tested - CookieStore not available';
```

Comment:
This results in some logic errors when the result are parsed and displayed because it's not `=== false`. It also messed up the downloaded results.

On FF and Safari you'll see:
```
first-party-sets-api - ✅ API available ("❌ Can't be tested - CookieStore not available")
```

Reply:
Yeah, good call I changed it to thrown exception instead - this way the output looks correct. See screenshot in other comment.